### DISCUSSION TITLE
Using shared vue components

### DISCUSSION THREAD

**Discussion by DavidPLamas** [directus/directus#11297] [Archive]
Hey everyone,

I'm currently developing a custom extension module and I would like to know if it is possible at all to use vue components that are currently being used in the main application. For instance, I would like to re-use the `navigation-item` but I haven't found a way to import that in my project. Maybe moving that to the @directus/shared package would make it usable but I'm not sure you would want to do that. 

Have you ever thought of of ways for providing this?

**Comment by rijkvanzanten** [comment]
All the `v-*` base components are available for use in extensions. See https://github.com/directus/directus/wiki/app-components for all available ones. They're all globally registered, so can be used without any imports 👍🏻

**Reply by DavidPLamas** [reply]
That's exactly what I was looking for. Thank you so much for the quick reply!

**Reply by eXsiLe95** [reply]
Hey :) 
Is there any possibility to register those components in the IDE to get support? I was aware that those component can be used, but development is very hard when (e.g.) IntelliJ doesn't know about those components at all...

**Reply by rijkvanzanten** [reply]
Hmm I don't think that's currently possible. Those components live in the app codebase source files, which aren't distributed in the `@directus/app` package that's on npm. The only alternative would be to pull them in from source `directus/directus`, but that isn't ideal 🤔
